xds/src/test/java/io/grpc/xds/WeightedRoundRobinLoadBalancerTest.java

```diff
@@ -821,45 +822,275 @@ public void run() {
+    sss.pick();
+  }
+
+  @Test
+  public void testPicksEqualsWeights() {
+    float[] weights = {1.0f, 2.0f, 3.0f};
+    Random random = new Random();
+    StaticStrideScheduler sss = new StaticStrideScheduler(weights, random);
+    int[] expectedPicks = new int[] {1, 2, 3};
+    int[] picks = new int[3];
+    for (int i = 0; i < 6; i++) {
+      picks[sss.pick()] += 1;
+    }
+    assertThat(picks).isEqualTo(expectedPicks);
+  }
+
+  @Test
+  public void testContainsZeroWeightUseMean() {
+    float[] weights = {3.0f, 0.0f, 1.0f};
+    Random random = new Random();
+    StaticStrideScheduler sss = new StaticStrideScheduler(weights, random);
+    int[] expectedPicks = new int[] {3, 2, 1};
+    int[] picks = new int[3];
+    for (int i = 0; i < 6; i++) {
+      picks[sss.pick()] += 1;
+    }
+    assertThat(picks).isEqualTo(expectedPicks);
+  }
+
+  @Test
+  public void testContainsNegativeWeightUseMean() {
+    float[] weights = {3.0f, -1.0f, 1.0f};
+    Random random = new Random();
+    StaticStrideScheduler sss = new StaticStrideScheduler(weights, random);
+    int[] expectedPicks = new int[] {3, 2, 1};
+    int[] picks = new int[3];
+    for (int i = 0; i < 6; i++) {
+      picks[sss.pick()] += 1;
+    }
+    assertThat(picks).isEqualTo(expectedPicks);
+  }
+
+  @Test
+  public void testAllSameWeights() {
+    float[] weights = {1.0f, 1.0f, 1.0f};
+    Random random = new Random();
+    StaticStrideScheduler sss = new StaticStrideScheduler(weights, random);
+    int[] expectedPicks = new int[] {2, 2, 2};
+    int[] picks = new int[3];
+    for (int i = 0; i < 6; i++) {
+      picks[sss.pick()] += 1;
+    }
+    assertThat(picks).isEqualTo(expectedPicks);
+  }
+
+  @Test
+  public void testAllZeroWeightsUseOne() {
+    float[] weights = {0.0f, 0.0f, 0.0f};
+    Random random = new Random();
+    StaticStrideScheduler sss = new StaticStrideScheduler(weights, random);
+    int[] expectedPicks = new int[] {2, 2, 2};
+    int[] picks = new int[3];
+    for (int i = 0; i < 6; i++) {
+      picks[sss.pick()] += 1;
+    }
+    assertThat(picks).isEqualTo(expectedPicks);
+  }
+
+  @Test
+  public void testAllInvalidWeightsUseOne() {
+    float[] weights = {-3.1f, -0.0f, 0.0f};
+    Random random = new Random();
+    StaticStrideScheduler sss = new StaticStrideScheduler(weights, random);
+    int[] expectedPicks = new int[] {2, 2, 2};
+    int[] picks = new int[3];
+    for (int i = 0; i < 6; i++) {
+      picks[sss.pick()] += 1;
+    }
+    assertThat(picks).isEqualTo(expectedPicks);
+  }
+
+  @Test
+  public void testLacrgestWeightIndexPickedEveryGeneration() {
+    float[] weights = {1.0f, 2.0f, 3.0f};
+    int mean = 2;
+    Random random = new Random();
+    StaticStrideScheduler sss = new StaticStrideScheduler(weights, random);
+    int largestWeightPickCount = 0;
+    int kMaxWeight = 65535;
+    for (int i = 0; i < mean * kMaxWeight; i++) {
+      if (sss.pick() == mean) {
```

Comment:
I can understand that you wanted to test that when the `pick()` returns the index of the largest weight, you increment the counter. But I don't understand why you name the local variable `mean` - it looks like you are comparing the `pick()` result with the mean of the weight, but actually not. So if you rename the `mean` in this comparison to `largestWeightIndex`, it will make sense to me.